src/app/frontend/common/components/middleellipsis/middleellipsis_directive.js

```diff
@@ -0,0 +1,65 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import MiddleEllipsisController from './middleellipsis_controller';
+import computeTextLength from './middleellipsis';
+
+/**
+ * Returns directive definition for middle ellipsis.
+ *
+ * @param {!angular.$filter} middleEllipsisFilter
+ *
+ * @return {!angular.Directive}
+ * @ngInject
+ */
+export default function middleEllipsisDirective(middleEllipsisFilter) {
+  return {
+    controller: MiddleEllipsisController,
+    controllerAs: 'ellipsisCtrl',
+    templateUrl: 'common/components/middleellipsis/middleellipsis.html',
+    scope: {},
+    /**
+     * @param {!angular.Scope} scope
+     * @param {!angular.JQLite} jQLiteElem
+     * @param {!angular.Attributes} attr
+     * @param {!MiddleEllipsisController} ctrl
+     */
+    link: function(scope, jQLiteElem, attr, ctrl) {
+      /** @type {!Element} */
+      let element = jQLiteElem[0];
+      let container = element.querySelector('.kd-middleellipsis-container');
+      let ellipsisElem = element.querySelector('.kd-middleellipsis');
+
+      if (!container) {
+        throw new Error(
+            `Required parent element with class .kd-middleellipsis-container not found`);
+      }
+
+      if (!ellipsisElem) {
+        throw new Error('Required element with class .kd-middleellipsis-nowrap not found');
+      }
+
+      let nonNullContainer = container;
+      let nonNullElement = ellipsisElem;
+
+      scope.$watch(
+          () => computeTextLength(nonNullContainer, nonNullElement, middleEllipsisFilter),
+          (newLength, oldLength) => { ctrl.maxLength = oldLength; });
```

Comment:
The problem is that string has 35 characters and algorithm returns 33 as the expected length. Truncated string will have actually 36 characters because there are `...` added from filter. I think this is a problem.